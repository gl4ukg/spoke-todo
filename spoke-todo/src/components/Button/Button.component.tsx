import { FC } from "react"
import { PlusOutlined } from "@ant-design/icons"
import "./Button.scss"

interface Props {
    title: string
    isIcon: boolean
    onClick?: () => void
}

const Button = ({title, isIcon, onClick}: Partial<Props>) => {
    return (
        <button 
            data-testid="button-id"
            className="button"
            onClick={onClick}
        >
            {isIcon && <PlusOutlined />}
            <p>{title}</p>
        </button>
    )
}

export default Button