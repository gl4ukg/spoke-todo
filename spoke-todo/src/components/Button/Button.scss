.button {
    display: flex;
    height: fit-content;
    background-color: var(--main);
    padding: 10px 20px;
    border-radius: 20px;
    border: 0;
    position: absolute;
    bottom: -20px;
    left: 50%;
    transform: translateX(-50%);
    cursor: pointer;
    transition: background-color .8s;
    &:hover {
        background-color: var(--blue);
    }
    svg {
        fill: var(--white);
    }
    p {
        font-weight: 600;
        color: var(--white);
        margin: 0 0 0 6px;
    }
}