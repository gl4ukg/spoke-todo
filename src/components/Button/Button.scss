.button {
    display: flex;
    height: fit-content;
    background-color: rgb(158, 33, 216);
    padding: 10px 20px;
    border-radius: 20px;
    border: 0;
    position: absolute;
    bottom: -20px;
    left: 50%;
    transform: translateX(-50%);
    svg {
        fill: #fff;
    }
    p {
        font-weight: 600;
        color: #fff;
        margin: 0 0 0 6px;
    }
}