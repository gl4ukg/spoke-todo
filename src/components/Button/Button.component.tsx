import { FC } from "react"
import { PlusOutlined } from "@ant-design/icons"
import "./Button.scss"

interface Props {
    onClick: () => void
}

const Button: FC<Props> = (props:Props) => {
    return (
        <button 
            onClick={props.onClick}
            className="button">
            <PlusOutlined />
            <p>New task</p>
        </button>
    )
}

export default Button