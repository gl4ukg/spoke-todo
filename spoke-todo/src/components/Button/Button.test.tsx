import { render, screen } from '@testing-library/react';
import Button from './Button.component';

describe('button', () => {
    it('should render with title prop correctly', () => {
        const { container } = render(<Button 
            title='Create'
            isIcon={true}
        />);

        expect(container).toBeInTheDocument();
        expect(container).toHaveTextContent('Create');
    })
    it('should render with isIcon prop correctly', () => {
        const { container } = render(<Button 
            title='Create'
            isIcon={true}
        />);

        expect(container).toBeInTheDocument();
        // expect(container).toHave
    })
    it('should render a button class of button', () => {
        render(<Button 
            title='Create'
            isIcon={true}
            />)
        expect(screen.getByTestId('button-id')).toHaveClass('button')
    })
    // it('should have img element when isIcon prop is present with isIcon as src and alt', () => {
    //     const { container } = render(<Button 
    //         text='Create'
    //         isIcon='photo'
    //     />);
    //     const imgElement = container.getElementsByTagName('img')[0];


    //     expect(container).toBeInTheDocument();
    //     expect(imgElement).toHaveAttribute('src', 'photo');
    //     expect(imgElement).toHaveAttribute('alt', 'photo');
    // })

    // it('should have given class name when class name prop is present', () => {
    //     const { container } = render(<Button 
    //         title='Create'
    //         isIcon={false}
    //         className='button'
    //     />);
        
    //     expect(container).toBeInTheDocument();
    //     expect(container.firstChild).toHaveClass('button');
    // })
});


describe('button snapshots', () => {
    it('should render correctly with title value', () => {
        const { container } = render(<Button title='Create' isIcon={false}/>)

        expect(container.firstChild).toMatchSnapshot()
    });
})