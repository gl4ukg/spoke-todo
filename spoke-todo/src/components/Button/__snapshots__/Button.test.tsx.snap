// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button snapshots should render correctly with title value 1`] = `
<button
  class="button"
  data-testid="button-id"
>
  <p>
    Create
  </p>
</button>
`;
